### PR TITLE
Cpp input

### PR DESCRIPTION
**Pull request by amrondonp**
Changes:

Back-End
nodejs service copaki.js now receives the user_input in the URL request as an array of strings
copaki.js calls run_cpp_backend.py with the input as a command line parameter

Front-End
iframe tool (iframe-embed.html and iframe-embed.js) now does not ignore the user input sent in the URL
The endpoint for calling the cpp service has changed so that it calls localhost

### CODE REVIEW

File: v3/js/opt-frontend-common.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ if (window.location.protocol === 'https:') {
   var JAVA_JSONP_ENDPOINT = 'http://104.237.139.253:3000/exec_java_jsonp'; // for deployment
   var RUBY_JSONP_ENDPOINT = 'http://104.237.139.253:3000/exec_ruby_jsonp'; // for deployment
   var C_JSONP_ENDPOINT = 'http://104.237.139.253:3000/exec_c_jsonp'; // for deployment
-  var CPP_JSONP_ENDPOINT = 'http://104.237.139.253:3000/exec_cpp_jsonp'; // for deployment
+  var CPP_JSONP_ENDPOINT = "http://127.0.0.1:3001/exec_cpp_jsonp"; // for deployment
```
Comment (0xDCA) [inline v3/js/opt-frontend-common.js]: Why does this use port 3001 instead of 3000?

File: v4-cokapi/cokapi.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 var IS_DEBUG = false;
 
-var PRODUCTION_PORT = 3000;
+var PRODUCTION_PORT = 3001;
```
Comment (0xDCA) [inline v4-cokapi/cokapi.js]: Why was this changed?

File: v3/js/iframe-embed.js [hunk 3]
```diff
@@ -63,6 +75,7 @@ $(document).ready(function() {
   var heapPrimitivesBool = (queryStrOptions.heapPrimitives == 'true');
   var textRefsBool = (queryStrOptions.textReferences == 'true');
   var cumModeBool = (queryStrOptions.cumulative == 'true');
+  rawInputLst = queryStrOptions.rawInputLst;
```
Comment (0xDCA) [inline v3/js/iframe-embed.js:78]: Please use local variables (var) or declare this as a global variable explicitly

Comment (0xDCA) [reply v3/js/iframe-embed.js:78]: Also. Why is this a list? Can we call it rawInputList instead of rawInputLst?

Comment (amrondonp) [reply v3/js/iframe-embed.js:78]: (as described by PythonTutor people)  This is an "ugly" variable that needs "refactoring" This global variable is used by several modules to handle user input, there is not passing information between functions but instead all of them have access to it. I agree that it need refactoring but I don't know how hard it would be

File: v4-cokapi/cokapi.js (hunk 2 above)
Comment (amrondonp) [reply v4-cokapi/cokapi.js]: Ubuntu 16.04 has a system process running by default in port 3000. So I couldn't run it unless I changed the port. As we maybe deploying in some similar system maybe it caused problems. However now that I see that it would be better to put the port as a console parameter (With the issue of having to change the fronted) What do you think

File: v3/js/iframe-embed.js (hunk 3 above)
Comment (0xDCA) [reply v3/js/iframe-embed.js:78]: Ok, that's fine then

File: v4-cokapi/cokapi.js (hunk 2 above)
Comment (0xDCA) [reply v4-cokapi/cokapi.js]: Good point. Yes, this kind of things should be passed as an argument to the program. Why do we need to change the frontend? I'm pretty sure you can set a default value.

Comment (amrondonp) [reply v4-cokapi/cokapi.js]: Because if the runner of this service changes the port then he would need to change the fronted URL and port too.

File: v3/js/opt-frontend-common.js (hunk 1 above)
Comment (amrondonp) [reply v3/js/opt-frontend-common.js]: @0xDCA We would need to change this specific line to match with the one that the runner of the service put. In that case passing the port as a parameter would cause more problems that solutions?

**Comment by 0xDCA** [comment]
LGTM
